src/GeositeFramework/js/Export.js

```diff
@@ -124,13 +154,31 @@ require(['use!Geosite'],
 
                     mapReadyDeferred.resolve();
                 },
-                closejs: function() {
-                    var mapNode = $("#map-0").detach()[0];
-                    mapNodeParent.append(mapNode);
-                    $("#export-print-preview-map").detach().appendTo($("#export-print-preview-container"));
+                closejs: function () {
                     $('.app-print-css').remove();
                     $('.print-orientation-css').remove();
+
                     $("#legend-container-0").css({ visibility: "visible" });
+                    _.delay(restoreCssDeferred.resolve, 500);
```

Comment:
It's unfortunate that we have to rely so heavily on `delay` in order to have things updated in time, but the use of `Deferreds` makes it fairly understandable.  I assume the result of not using it is that the css would not be applied and rendered by the time the print preview is displayed?  In both the close and open functions, do the delays need to be incremental, or can they be accumulated into a single delay (ie, all the properties are set, css loaded, etc - then a delay - then the print call)?